### PR TITLE
Fix OSX warnings

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by zzmp** [comment]
@Atlante45 [you added the pragma against -Wpessimizing-move](https://github.com/highfidelity/hifi/commit/eb21b9c5d778675bbc431f1377d2b03ff7bf9072). Is it necessary? I am running `XCode 7.1.1 (7B1005)` and it gave me a warning.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 11e773dd1aa1db4a9c57a7d9e1d766bf19795de2 available at: https://gist.github.com/81b1f1b27d6100459dfd684f0968ba92<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7919.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7919.dmg

**Comment by Atlante45** [comment]
@zzmp It is, but if you look a few lines lower I pop the change. It's meant to disable it only while we process the oglplus headers we have no control over and that produce that warning.
If you see that warning somewhere else in our code, then we should just fix it.
Nothing we can do for external libraries except hiding it.

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wpessimizing-move"

// external headers, we ignore pessimizing move warnings
		 
#pragma clang diagnostic pop
```

File: libraries/gl/src/gl/OglplusHelpers.h [hunk 1]
```diff
@@ -30,7 +30,6 @@
 #pragma warning( disable : 4068 )
 #elif defined(Q_OS_MAC)
 #pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wpessimizing-move"
```
Comment (Atlante45) [inline libraries/gl/src/gl/OglplusHelpers.h]: This is not the cause of the warning, this hides those warning for external library headers.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e1804802b7cdb964c4e17c257f5012c28ba2777 available at: https://gist.github.com/81b1f1b27d6100459dfd684f0968ba92<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7919.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7919.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 59e17c2d5f66e801c889349f114cb3c154e4e54e available at: https://gist.github.com/81b1f1b27d6100459dfd684f0968ba92<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7919.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7919.dmg

**Comment by zzmp** [comment]
#### Testing

Smoke test.
This was to fix compiler warnings, not to fix a specific feature, so a smoke test should be sufficient, so long as it covers things that glm would cover: rendering and physics.

**Comment by OmegaHeron** [comment]
QA Pass
Tested Windows 10 64 bit - i5 6500 CPU - GTX970 GPU - **Desktop mode - keyboard/mouse**
All standard tests pass.  Mic/cam icons not located correctly, but that is not this PR's issue - #7908 references cause..

**Pull request merged**